app/helpers/automated_tests_helper.rb

```diff
@@ -48,15 +48,21 @@ def update_test_groups_from_specs(assignment, test_specs)
         display_output = extra_data_specs['display_output'] || TestGroup.display_outputs.keys.first
         test_group_name = extra_data_specs['name'] || TestGroup.model_name.human
         criterion_id = nil
-        unless extra_data_specs['criterion'].nil?
-          criterion_id = extra_data_specs['criterion']
+        unless extra_data_specs['criterion_name'].nil?
```

Comment:
You should not need to change this method (see the comment for the `strip_ids_from_spec` method)